app/src/androidTest/java/com/example/sharingang/MapFragmentTest.kt

```diff
@@ -1,24 +1,38 @@
 package com.example.sharingang
 
+import android.Manifest
 import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.click
 import androidx.test.espresso.assertion.ViewAssertions.matches
 import androidx.test.espresso.matcher.ViewMatchers.withId
 import androidx.test.espresso.matcher.ViewMatchers.withText
 import androidx.test.ext.junit.rules.ActivityScenarioRule
 import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.rule.GrantPermissionRule
+import dagger.hilt.android.testing.HiltAndroidRule
+import dagger.hilt.android.testing.HiltAndroidTest
 import org.hamcrest.core.StringContains.containsString
 import org.junit.Rule
 import org.junit.Test
 import org.junit.runner.RunWith
 
 
 @RunWith(AndroidJUnit4::class)
-class MapActivityTest {
+@HiltAndroidTest
+class MapFragmentTest {
+    @get:Rule(order = 0)
+    val hiltRule = HiltAndroidRule(this)
+
+    @get:Rule(order = 1)
+    val activityRule = ActivityScenarioRule(MainActivity::class.java)
+
     @get:Rule
-    val activityRule = ActivityScenarioRule(MapActivity::class.java)
+    val grantPermissionRule: GrantPermissionRule =
+        GrantPermissionRule.grant(Manifest.permission.ACCESS_FINE_LOCATION)
 
     @Test
-    fun grantingPermissionDisplaysLocation() {
+    fun openingFragmentDisplaysLocation() {
+        onView(withId(R.id.go_to_map)).perform(click())
         val text = onView(withId(R.id.location_display))
         text.check(matches(withText("")))
         Thread.sleep(10000)
```

Comment:
Do we really have to wait this long? Is there not a better way here?

Reply:
I probably can reduce it to 5 seconds, but I wasn't able to find a way to wait for the location result...